tests/copy-paste.js

```diff
@@ -0,0 +1,780 @@
+                ["/copy-1-of-text", "Text", "text2", "text2", "text"],
+            ]);
+        });
+
+        it("should paste two questions with the same id (one in group)", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "text", "text"],
+                ["/group", "Group", "group", "group"],
+                ["/group/text", "Text", "text", "text"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "text", "text"],
+                ["/group", "Group", "group", "group"],
+                ["/group/text", "Text", "text", "text"],
+            ]);
+        });
+
+        it("should paste three groups after text question", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "text", "text"],
+            ]);
+            util.clickQuestion("text");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/group1", "Group", "group", "group"],
+                ["/group2", "Group", "group", "group"],
+                ["/group3", "Group", "group", "group"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "text", "text"],
+                ["/group1", "Group", "group", "group"],
+                ["/group2", "Group", "group", "group"],
+                ["/group3", "Group", "group", "group"],
+            ]);
+        });
+
+        it("should paste three groups into group", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/group", "Group", "group", "group"],
+            ]);
+            util.clickQuestion("group");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/group1", "Group", "group", "group"],
+                ["/group2", "Group", "group", "group"],
+                ["/group3", "Group", "group", "group"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/group", "Group", "group", "group"],
+                ["/group/group1", "Group", "group", "group"],
+                ["/group/group2", "Group", "group", "group"],
+                ["/group/group3", "Group", "group", "group"],
+            ]);
+        });
+
+        it("should resolve pasted question with conflicting question ID", function () {
```

Comment:
This happens if you copy a question with a conflicted question ID, and then paste that somewhere where the question ID no longer has a conflict. It verifies that the conflict is resolved on paste.